src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -159,6 +159,43 @@ public ComplaintResponse createComplaintResponseRepresentingLock(Complaint compl
         return persistedComplaintResponse;
     }
 
+    /**
+     * Resolves a complaint by filling in the empty complaint response attached to it
+     *
+     * The empty complaint response acts as a lock. Only the creator of the empty complaint response and instructors can resolve empty complaint response as long as the lock
+     * is running. For lock duration calculation see: {@link ComplaintResponse#isCurrentlyLocked()}. These methods fill in the initial complaint response and either accepts
+     * or denies the associated complaint, thus resolving the complaint
+     *
+     * @param updatedComplaintResponse complaint response containing the information necessary for resolving the complaint
+     * @param complaintResponseId      complaint response id
+     * @return complaintResponse of resolved complaint
+     */
+    public ComplaintResponse resolveComplaint(ComplaintResponseUpdateDTO updatedComplaintResponse, Long complaintResponseId) {
+        validateComplaintResponseId(complaintResponseId);
+        ComplaintResponse complaintResponseFromDatabase = fetchComplaintResponseOrThrow(complaintResponseId);
+        validateComplaintResponseEmpty(complaintResponseFromDatabase);
+        Complaint originalComplaint = fetchOriginalComplaintOrThrow(complaintResponseFromDatabase);
```

Comment:
As this affects the other method too, and refactoring is out of scope, please add a TODO to make this retrieval redundant by proper fetching